Advanced Search

kelcol opened the pull request:
For issue #4, I added a form with an input for text and a controlled field so users can search for poems filtering by genre. More controlled fields can be added (date ranges, length, etc.) later but this is a start.  


I put this form on the home page under the jumbotron, since I think anything more than a basic search form would be too busy to put in the nav bar. Not sure about this location though, since it takes up a bit of prime real-estate :grimacing: What do you all think about creating a dedicated page for conducting advanced searches? Another option might be to add this form as a collapsing/expanding component on the home page.

Commit:
Added page for listing content providers

Commit:
Removed placeholder text

Commit:
Add metadata planning doc

Commit:
Added a search-by-genre form

masonpam commented:
Maybe have the genre, and more advanced, search form would be better as an expanding/collapsing component on the homepage.  That way it doesn't take up too much space away from the features.  What do you think?

masonpam commented:
I tested out your code for the genre search form, it all works great, I think that would be a good feature to add to the navigation or maybe an advanced search feature.  I do think that you could add a submit button around line 70 and add a blank field in the list of genres, that way if the result isn't in the drop-down users can still search on their own.  

![screenshot 70](https://user-images.githubusercontent.com/26751847/43681059-cf36f1ba-97fd-11e8-8868-ff521c84ab04.png)

csdarcher commented:
Code looks great! I agree that a submit button is needed. If we add an advanced search option in the future, what do you all think about having that form pop open like a modal?

Commit:
Made adv search form a modal div; added none-selected as genre search option

kelcol commented:
I added a "search" button to the genre/advanced search form. Thanks for catching that @masonpam! I also added a 'None Selected' entry for the controlled genre field. 

I implemented @csdarcher's idea to make the advanced search form a modal window in order to keep it from cluttering up the home page. We can add an "Advanced Search" link right next to the "Basic Search" in the navbar. I think that would be the most intuitive place to put it.

Commit:
Added none-selected as genre option; title-cased genre options displayed in dropdown

beigid commented on index.html:
```diff
@@ -60,6 +60,57 @@ <h1>Hello, world!</h1>
+        <div class="col-md-4">
+          <h2>Search for Poetry by Genre</h2>
+          <form role="search" action="submit">
+              <label for="search-field">
```
I think having a search field is a great idea! The search by genre form will allow the project to be easily narrowed down to an area or specific poem a user might be looking to filter through. Good use of using label and the form HTML tags for this!

beigid reviewed (commented):
I think having a search field is a great idea! The search by genre form will allow the project to be easily narrowed down to an area or specific poem a user might be looking to filter through. Good use of using label and the form HTML tags for this!

masonpam commented:
I tested out the code in my environment.  It seems like the search bar that was created is gone.  

![screenshot 78](https://user-images.githubusercontent.com/26751847/43972976-f78fc3b2-9c8a-11e8-92d5-9f033173acb6.png)

I can see the code under the jumbotron starting at line 64.  But it isn't showing up when I preview it.  

![screenshot 79](https://user-images.githubusercontent.com/26751847/43973026-1a5bd76e-9c8b-11e8-8bb8-4624897c4572.png)

I tested a couple different things and when I got rid of <code>class="modal fade"</code> on line 65 the modal came right back.  I think there was an error with code not recognizing spaces.  So the <code>class="modal fade"</code> from line 65 should actually look like <code>class="modal-fade"</code>.  I tested this out too and the search area is there when I changed that.  
I think your code looks great and with that small change it's ready to merge and deploy.

kelcol commented:
Adjusted title of pull request to be more descriptive: Content metadata => Advanced Search

Commit:
Add basic and advanced search to nav bar

Commit:
style search in nav

Commit:
remove duplicate id from modal

kelcol commented:
Thanks for the review, ya'll! I've added basic and advanced search to the nav bar and the advanced search modal is visible/working now. @masonpam, thanks for testing out the modal. The `modal` and `fade` classes seem to be getting properly applied now.
